cirq-core/cirq/contrib/quimb/mps_simulator.py

```diff
@@ -440,9 +440,16 @@ def apply_op(self, op: 'cirq.Operation', prng: np.random.RandomState):
             raise ValueError('Can only handle 1 and 2 qubit operations')
         return True
 
-    def _act_on_fallback_(self, op: Any, qubits: Sequence['cirq.Qid'], allow_decompose: bool):
+    def _act_on_fallback_(
+        self,
+        action: Union['cirq.Operation', 'cirq.Gate'],
+        qubits: Sequence['cirq.Qid'],
+        allow_decompose: bool = True,
+    ):
         """Delegates the action to self.apply_op"""
-        return self.apply_op(op, self.prng)
+        if isinstance(action, ops.Gate):
+            action = ops.GateOperation(action, qubits)
```

Comment:
There's already an apply_op in the MPS simulator that has this signature, so this was the least invasive fix.

I tried a couple months ago to separate act_on into two distinct protocols, one for ops and one for gates, but that got met with some resistance. So now there's some flopping back and forth between gates and ops within implementations of the protocol. I'll spend a little time looking into whether there's a better way to consolidate. It may make sense to do this promotion right as the first step of protocols.act_on itself. But obviously outside the scope of this PR.

Reply:
I opened it as a new issue in https://github.com/quantumlib/Cirq/issues/4400